back/boxtribute_server/business_logic/metrics/queries.py

```diff
@@ -16,3 +20,70 @@ def resolve_metrics(*_, organisation_id=None):
 
     # Pass organisation ID to child resolvers
     return {"organisation_id": organisation_id}
+
+
+@public_query.field("newlyRegisteredBeneficiaryNumbers")
+def resolve_newlyRegisteredBeneficiaryNumbers(*_):
+    now = datetime.now()
+
+    # Last month
+    first_day_this_month = now.replace(day=1)
+    last_day_of_last_month = first_day_this_month - timedelta(days=1)
+    first_day_of_last_month = last_day_of_last_month.replace(day=1)
+
+    # Last quarter
+    curr_quarter = (now.month - 1) // 3 + 1
+    last_quarter = curr_quarter - 1
+
+    year = now.year
+    if last_quarter == 0:
+        last_quarter = 4
+        year -= 1
+
+    first_month_last_quarter = 3 * (last_quarter - 1) + 1
+    quarter_start = date(year, first_month_last_quarter, 1)
+
+    if last_quarter == 4:
+        next_quarter_start = date(year + 1, 1, 1)
+    else:
+        next_quarter_start = date(year, first_month_last_quarter + 3, 1)
+
+    quarter_end = next_quarter_start - timedelta(days=1)
+
+    # Last year
+    start_last_year = date(now.year - 1, 1, 1)
+    end_last_year = date(now.year - 1, 12, 31)
```

Comment:
Just a naming consistency remark: please use one of `first_day_of_last_X` or `start_last_X` for X in (month, quarter, year), same for `last_day`/`end_last`